File: licenses-project-api-main/GSLP.Application/Services/Platform/LicencaService/DTOs/LicencaDTO.cs
using GSLP.Application.Common.Marker;
using GSLP.Application.Services.Application.AplicacaoService.DTOs;
using GSLP.Application.Services.Platform.ClienteService.DTOs;
using GSLP.Application.Services.Platform.LicencaFuncionalidadeService.DTOs;
using GSLP.Application.Services.Platform.LicencaModuloService.DTOs;
using GSLP.Application.Services.Platform.PerfilService.DTOs;

namespace GSLP.Application.Services.Platform.LicencaService.DTOs
{
    public class LicencaDTO : IDto
    {
        public string? Id { get; set; }
        public string? Nome { get; set; }
        public DateTime DataInicio { get; set; }
        public DateTime DataFim { get; set; }
        public int NumeroUtilizadores { get; set; }
        public bool Ativo { get; set; }
        public bool Bloqueada { get; set; }
        public DateTime? DataBloqueio { get; set; }
        public string? MotivoBloqueio { get; set; }
        public string? AplicacaoId { get; set; }
        public AplicacaoDTO Aplicacao { get; set; }
        public string? ClienteId { get; set; }
        public ClienteBasicDTO Cliente { get; set; }
        public ICollection<LicencaModuloDTO> LicencasModulos { get; set; } = [];
        public ICollection<LicencaFuncionalidadeDTO> LicencasFuncionalidades { get; set; } = [];
        public ICollection<PerfilBasicDTO> Perfis { get; set; } = [];
        public DateTime CreatedOn { get; set; }
        public string? APIKey { get; set; }
        public string? VersaoInstalada { get; set; } // Current installed version
        public string? FrontendPath { get; set; } // Path to frontend application
        public string? ApiPath { get; set; } // Path to API application
        public string? ApiPoolName { get; set; } // IIS Application Pool name for API
        public string? FrontendPoolName { get; set; } // IIS Application Pool name for Client/Frontend
        public string? Url1 { get; set; }
        public string? Url2 { get; set; }
        public string? Url3 { get; set; }
        public string? Url4 { get; set; }
        public string? Url5 { get; set; }
        public string? Url6 { get; set; }
        public string? Url7 { get; set; }
        public string? Url8 { get; set; }
        public bool? UseOwnUpdater { get; set; }
    }
}

File: licenses-project-frontend-main/src/types/dtos/licenca.dto.ts
export interface CreateLicencaDTO {
  nome: string
  dataInicio?: Date
  dataFim?: Date
  numeroUtilizadores: number
  aplicacaoId: string
  clienteId: string
  useOwnUpdater?: boolean
  frontendPath?: string
  apiPath?: string
  apiPoolName?: string
  frontendPoolName?: string
  url1?: string
  url2?: string
  url3?: string
  url4?: string
  url5?: string
  url6?: string
  url7?: string
  url8?: string
}

export interface UpdateLicencaDTO extends CreateLicencaDTO {
  id?: string
  ativo: boolean
}

export interface LicencaDTO {
  id: string
  nome: string
  dataInicio?: Date
  dataFim?: Date
  numeroUtilizadores: number
  ativo?: boolean
  aplicacaoId: string
  bloqueada?: boolean
  dataBloqueio?: Date
  motivoBloqueio?: string
  versaoInstalada?: string
  useOwnUpdater?: boolean
  frontendPath?: string
  apiPath?: string
  apiPoolName?: string
  frontendPoolName?: string
  url1?: string
  url2?: string
  url3?: string
  url4?: string
  url5?: string
  url6?: string
  url7?: string
  url8?: string
  aplicacao?: {
    nome: string
    versao?: string
    area?: {
      nome: string
      color: string
    }
  }
  cliente?: {
    nome: string
  }
  clienteId: string
  licencasFuncionalidades?: LicencaFuncionalidadeDTO[]
  licencasModulos?: LicencaModuloDTO[]
  apiKey: string
}

export interface LicencaFuncionalidadeDTO {
  licencaId: string
  funcionalidadeId: string
}

export interface LicencaModuloDTO {
  licencaId: string
  moduloId: string
}

export interface LicencaAPIKeyDTO {
  apiKey: string
  ativo: boolean
}

export interface BloqueioLicencaDTO {
  motivoBloqueio: string
}

export interface ModuloFuncionalidadeDTO {
  moduloId: string
  funcionalidadeId: string
}
